src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
I've updated the PR to unlock the registers after `genExitCode` for blocks that end in a `JMP`.